docs: add Hosting your own Chat Relay guide (#7263)
simplex-chat/simplex-chat: docs/CHAT-RELAY.md

---
title: Hosting your own Chat Relay
revision: 16.07.2026
---

# Hosting your own Chat Relay

> A chat relay is the SimpleX Chat CLI (`simplex-chat`) running in relay mode (`--relay`). It has its own profile (a display name and a picture), its own address, and it can serve web preview pages for the channels it hosts.

Reviewer: ```suggestion
Chat relays are used to deliver channel messages in SimpleX Network. Read more about channels in this [whitepaper](https://github.com/simplex-chat/simplex-chat/blob/master/docs/protocol/channels-overview.md) and this [blog post](https://simplex.chat/blog/20260430-simplex-channels-v6-5-consortium-crowdfunding-freedom-of-speech.html).

A chat relay is the SimpleX Chat CLI (`simplex-chat`) running in relay mode (`--relay`). It has its own profile (a display name and a picture), its own address, and in addition to delivering messages, it can generate data for web previews of the channels it delivers.
```

> This guide sets up a relay on a Linux server, runs it, and (optionally) serves channel web previews behind [Caddy](https://caddyserver.com).

Reviewer: ```suggestion
This guide explains how to set up a chat relay on a Linux server, how to run it, and (optionally) how to configure [Caddy](https://caddyserver.com) to serve data for channel web previews.
```

> **This guide applies only to SimpleX Chat v7.0.0-beta.4, which is not yet released.** The flags and behaviour it describes are not available in earlier versions.

> 

Reviewer: ```suggestion
> **Please note**: This guide applies only to SimpleX Chat v7.0.0-beta.4 and later.

```

| `--user-image-file FILE` | The relay's picture, from a `.png`, `.jpg` or `.jpeg` file. Applied **only when the profile is created**; ignored afterwards. |
| `--relay-address-server SERVER` | Create the relay address on a specific SMP server, e.g. `smp://<fingerprint>@smp.example.com`. By default a preset server is used. Requires `--relay`. |

### Get the relay address

The address is created and logged on the first start. Read it from the journal at any time:

```sh
journalctl -u simplex-relay | grep -A1 "address is created"
```

### Run relay commands

The service runs headless, so there is no attached terminal to type into. To run a one-off command, stop the service, run the command against the relay's database with `-e`, then start it again. For example, to change the picture (`--user-image-file` only sets it when the profile is first created):

```sh
systemctl stop simplex-relay
simplex-chat-relay -d /home/relay/relay -e "/set profile image file /home/relay/new-avatar.png"
systemctl start simplex-relay
```

## Channel web previews

> A relay can render recent messages of its public channels as JSON files, which you serve over HTTPS so channel links open a web preview. This is optional.

Reviewer: ```suggestion
Chat relays can render recent messages of its public channels as JSON files, which can be served over HTTPS using a web server to create channel web previews. This is optional.
```
